R/pkg/R/install.R

```diff
@@ -0,0 +1,84 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Functions to install Spark in case the user directly downloads SparkR
+# from CRAN.
+
+#' Download and Install Spark to Local Directory
+#' 
+#' \code{install_spark} downloads and installs Spark to local directory if
+#' it is not found. The Spark version we use is 2.0.0 (preview). Users can
+#' specify a desired Hadoop version, the remote site, and the directory where
+#' the package is installed locally.
+#'
+#' @param hadoop_version Version of Hadoop to install. 2.3, 2.4, 2.6,
+#'        and 2.7 (default)
+#' @param url the base URL of the repositories to use
+#' @param local_dir local directory that Spark is installed to
+#' @rdname install_spark
+#' @name install_spark
+#' @export
+#' @examples
+#'\dontrun{
+#' install_spark()
+#'}
+#' @note install_spark since 2.1.0
+install_spark <- function(hadoop_version = NULL, url = NULL, local_dir = NULL) {
+  version <- paste0("spark-", spark_version_default())
+  hadoop_version <- hadoop_version_default()
```

Comment:
Thanks, I'll fix the overwriting issue and rudimentary check will also be added :) For me I slightly prefer to refer to some external source (e.g. csv file). In case any change is needed, we only have to change the file without touching the source code, and it would be easier if we are going to support multiple versions later? 

about this @mengxr and I had some discussion previously. We thought about maintaining a table online containing available mirror sites so that we can make changes when necessary without touching the R package.